### PR TITLE
OT-376

### PR DESCRIPTION
**Pull request by polishcode**
- moved docker development files to isolated location
- renamed docker images to matterfi sub-namespace
- added missing .sh extension to ci build scripts
- explicitly added libargon2 - required by build
  used to be added implicitly as transitive dependency
  but after optimization of dnf is not installed yet required
  in some builds
- updated ci for fedora 32:
  - optimized every dnf
  - optimized qt build
    - qt without -system-assimp
    - qt compiles only required deps
      - 20 minutes faster image build time
      - 110 MB off the image size
  - qt 5.15.2 -> 5.15.5
  - cmake 3.22.2 -> 3.23.2
  - opendht 2.3.2 -> 2.4.5
- added docker ci for fedora 33:
  - iwyu clang 11
- added docker ci for fedora 34:
  - iwyu clang 12
- added docker ci for fedora 35:
  - iwyu clang 13
- added docker ci for fedora 36:
  - iwyu clang 14
  - builds/tests fail with it

### CODE REVIEW

File: _devel/ci/build-in-container-clang.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+docker run \
+    --mount type=bind,src=/mnt/f/repo-matterfi/opentxs,dst=/home/src,readonly \
+    --mount type=bind,src=/home/pgawron/opentxs/output,dst=/home/output \
+    -i --entrypoint /usr/bin/build-opentxs-clang.sh \
+    polishcode/matterfi-ci-fedora:35-1 nopch
```
Comment (TomaszAugustyn) [inline _devel/ci/build-in-container-clang.sh:7]: Can we add an empty new line at the end of .sh scripts? Github is complaining about it. Same for files:
 - _devel/ci/build-in-container-gcc.sh
 - _devel/ci/run-container.sh
 - _devel/ci/test-in-container.sh
 - _devel/flatpak/run-container.sh

and also complains about this one:
_devel/flatpak/README

**Review by TomaszAugustyn — APPROVED** [review]

File: _devel/ci/run-container.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+docker run \
+    -it \
+    --mount type=bind,src=/mnt/f/repo-matterfi/opentxs,dst=/home/src,readonly \
+    --mount type=bind,src=/mnt/f/repo-matterfi/docker/_devel/ci,dst=/home/script,readonly \
+    --mount type=bind,src=/home/pgawron/opentxs/output,dst=/home/output \
```
Comment (TomaszAugustyn) [inline _devel/ci/run-container.sh:7]: Just wondering why in many places there is a path to your home directory?

File: _devel/ci/build-in-container-clang.sh (hunk 1 above)
Comment (polishcode) [reply _devel/ci/build-in-container-clang.sh]: fixed in second commit

File: _devel/ci/run-container.sh (hunk 2 above)
Comment (polishcode) [reply _devel/ci/run-container.sh:7]: These are utility/development files for local machine, thus allowed myself to have those paths.

**Pull request merged**